requirements/specifications/lifecycle/index.md

```diff
@@ -479,81 +682,255 @@ scope for this document.
+app that is not in the `RUNNING` state and the remainder of this section
 does not need to happen.
 
-At this point, the app **MUST** be in the inactive state.
+At this point, the app **MUST** be in the `RUNNING` state.
 
-Next, if the app has permission to use the
-xrn:firebolt:capability:lifecycle:deallocation capability **and** the
-app has registered its implementation of the
-Lifecycle.ResourceDeallocator interface, then the platform **MUST** call
-the apps ResourceDeallocator.unload method via the onRequestUnload RPC
-call and wait for either the onUnloadResponse or onUnloadError handshake
-before proceeding. The finished API has no effect for apps with a valid
-ResourceDeallocator implementation.
+If an app provides the `xrn:firebolt:capability:lifecycle:application`
+capability, then the platform **MUST** call the app's implementation of
+`Application.unload()`:
 
-Finally, if the app does not have a permitted ResourceDeallocator
-implementation, or if the app's implementation has responded with a
-success handshake,, the platform **MUST** transition the app to the
-unloading state (see '[Unloading](#25-unloading)' above).
+> The platform **MUST** dispatch the `Lifecycle.onRequestDestroy`
+> notification to the app, and wait for `appDestroyTimeout` milliseconds
+> for either a `Lifecycle.destroyResult` or `Lifecycle.destroyError`
+> call in response.
+>
+> Once the platform receives the `destroyResult` call, then the platform
+> may proceed with app unloading.
+>
+> If the app times out or makes an `destroyError` call, then the app
+> **MUST** be terminated.
 
-### 3.8. Terminating an app
+### 4.10. Terminating an app
 
 Terminating an app removes it from memory without dispatching any state
 changes.
 
-The platform may terminate apps for any number of reasons that are out
-of scope for this document, e.g. an app becomes non-responsive.
+The platform **SHOULD** terminate apps when one of the transitions outlined
+here fails for some reason, or if an app becomes non-responsesive.
 
 Terminating an app results in removing it from memory and all resources
 held by the app and it's container being freed up.
 
 Platforms **MAY** terminate an app when needed but **SHOULD NOT** do
-this in place of graceful unloading.
-
-## 4. Core SDK APIs
+this in place of graceful [unloading](#38-unloading-an-app).
+
+## 5. Getting the current status
+The Lifecycle module **MUST** provide a `status` property API that
+returns the current lifecycle transition of the app if one is in
+progress, otherwise, it returns the current state.
+
+If an app is in a transtition from one state to another, then it
+**MUST** be considered in the transition until
+such time as the app's implementation of the [transition](#4-lifecycle-state-transitions) has returned
+and the platfrom has finshed the transition.
+
+The `status` API **MUST** have a corresponding `onStatusChanged`
+notification.
+
+The `status` API must have one of the following values:
+
+- `Initializing`
```

Comment:
would it be easier to show the current state, and the new state instead?

Reply:
think I agree with above, instead of inventing new language for the transition just use the existing language for the start and end of the transition.